### PR TITLE
add validation for customresourcedefintions

### PR DESCRIPTION
**Pull request by deads2k**
Add basic validation for customresource definitions.

@adohe if you had review bandwidth, this is a relatively small one.

### CODE REVIEW

File: staging/src/k8s.io/kube-apiextensions-server/pkg/apis/apiextensions/validation/validation.go [hunk 1]
```diff
@@ -15,3 +15,137 @@ limitations under the License.
...
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kube-apiextensions-server/pkg/apis/apiextensions"
+)
+
+func ValidateCustomResource(obj *apiextensions.CustomResource) field.ErrorList {
+	nameValidationFn := func(name string, prefix bool) []string {
+		ret := genericvalidation.NameIsDNSSubdomain(name, prefix)
+		requiredName := obj.Spec.Names.Plural + "." + obj.Spec.Group
+		if name != requiredName {
+			ret = append(ret, fmt.Sprintf(`must be spec.names.plural+"."+spec.group `))
```
Comment (sttts) [inline staging/src/k8s.io/kube-apiextensions-server/pkg/apis/apiextensions/validation/validation.go]: no space after group

File: staging/src/k8s.io/kube-apiextensions-server/pkg/apis/apiextensions/validation/validation.go [hunk 2]
```diff
@@ -15,3 +15,137 @@ limitations under the License.
...
+
+	return allErrs
+}
+func ValidateCustomResourceStatus(status *apiextensions.CustomResourceStatus, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, ValidateCustomResourceNames(&status.AcceptedNames, fldPath.Child("acceptedNames"))...)
+	return allErrs
+}
+
+func ValidateCustomResourceNames(names *apiextensions.CustomResourceNames, fldPath *field.Path) field.ErrorList {
```
Comment (sttts) [inline staging/src/k8s.io/kube-apiextensions-server/pkg/apis/apiextensions/validation/validation.go]: Do you want to allow empty Kind and Plural? Plural could be derived from the CustomResource name, but Kind?

Comment (deads2k) [inline staging/src/k8s.io/kube-apiextensions-server/pkg/apis/apiextensions/validation/validation.go]: > Do you want to allow empty Kind and Plural? Plural could be derived from the CustomResource name, but Kind?

This used by status validation too.  Status may not have all these field set.  If they are set, they must be valid, but if they are not set how to handle that varies with where the field is embedded.  See the spec validation above.

Also, no information is derived post-defaulting: https://github.com/kubernetes/kubernetes/pull/45561

**Comment by deads2k** [comment]
its the api-extensions thing again: https://github.com/kubernetes/kubernetes/pull/45490 approving

**Pull request merged**